kv/kvserver: TestRaftTransportClockPropagation failed

kv/kvserver.TestRaftTransportClockPropagation [failed](https://tanzanite.cluster.engflow.com/invocations/default/447c62e8-ed0e-45cc-9109-0c6a171b9580?testReportRun=1&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [35511709ab438624dbf5cf69b5b6e64ee2b9948c](https://github.com/cockroachdb/cockroach/commits/35511709ab438624dbf5cf69b5b6e64ee2b9948c):


```
=== RUN   TestRaftTransportClockPropagation
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestRaftTransportClockPropagation2005851968
    test_log_scope.go:81: use -show-logs to present logs inline
    raft_transport_test.go:747: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/raft_transport_test.go:747
        	Error:      	Should be true
        	Test:       	TestRaftTransportClockPropagation
    panic.go:626: -- test log scope end --
--- FAIL: TestRaftTransportClockPropagation (0.01s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/replication
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRaftTransportClockPropagation.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39535

--- a/pkg/kv/kvserver/raft_transport_test.go
+++ b/pkg/kv/kvserver/raft_transport_test.go
@@ -741,10 +741,12 @@ func TestRaftTransportClockPropagation(t *testing.T) {
 	}
 
 	// Advance the client's clock beyond the server's.
-	rttc.clocks[clientReplica.NodeID].manual.Increment(1000)
-	clientNow := rttc.clocks[clientReplica.NodeID].clock.Now()
 	serverNow := rttc.clocks[serverReplica.NodeID].clock.Now()
-	require.True(t, serverNow.Less(clientNow))
+	var clientNow hlc.Timestamp
+	for clientNow.LessEq(serverNow) {
+		rttc.clocks[clientReplica.NodeID].manual.Increment(1000000)
+		clientNow = rttc.clocks[clientReplica.NodeID].clock.Now()
+	}
 
 	// Send a message from the client to the server.
 	sent := rttc.Send(clientReplica, serverReplica, 1 /* rangeID */, raftpb.Message{Commit: 10})

